### ISSUE TITLE
NSCache not returning object for key...

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/DiffableTextViews#57]
Cached instances are never found and a new one is created each update... (but at least there is a cache size limit).

**Comment by oscbyspro** [comment]
Dunno, neither the key's hash nor its equality method is called in nscache.object(forKey:)...

**Comment by oscbyspro** [comment]
It looks like NSCache is somehow reference based? It's stupid.

```swift
import Foundation

final class Key: Hashable {
    init() { }
    
    func hash(into hasher: inout Hasher) { }
    
    static func == (lhs: Key, rhs: Key) -> Bool { true }
}

final class Value {
    init() { }
}

let nscache = NSCache<Key, Value>()

let key0  = Key()
let key1  = Key()
let value = Value()

nscache.setObject(value,  forKey: key0)
let none = nscache.object(forKey: key1)
```

The above result in no object found, where as the object is found if `let key1 = key0`.

**Comment by oscbyspro** [comment]
Apparently the key must subclass NSObject, then override isEqual(_ object: Any?) and hash: Int. Thanks for not stating that anywhere. Saved by https://medium.com/anysuggestion/how-to-use-custom-type-as-a-key-for-nscache-9bdbee02a8f1.

**Comment by oscbyspro** [comment]
Fixed in a51dc7479c59e8ac98ff668355501ae90b296794 and will be included in v3.0.0.
